src/networking.c

```diff
@@ -1172,6 +1172,18 @@ void deferredAfterErrorReply(client *c, list *errors) {
     }
 }
 
+/* Reset the reply buffer and/or list of reply links.
+ * of the client to an empty state. */
+void resetReply(client *c) {
+    c->bufpos = 0;
+    c->reply_bytes = 0;
+    listEmpty(c->reply);
+    if (c->deferred_reply_errors) {
+        listRelease(c->deferred_reply_errors);
+        c->deferred_reply_errors = NULL;
+    }
+}
```

Comment:
this doesn't only resets a reply of the current command.
it resets the entire client, anything that wasn't transmitted yet.
i.e. multiple commands in case of pipeline.

but maybe even worse, it could cause protocol breakage, in case part of a command's reply was already transmitted, and part of it wasn't yet, so while you process the next command that arrives (if client uses pipeline), it will break the protocol on the client connection.

if we want to implement such an API, we must keep some marker of what's to reset.

Reply:
Good point.
Closing this for now.